Moab.Models/Tour_Trail.cs

```diff
@@ -31,7 +31,14 @@ public class Tour_Trail
         /// <returns></returns>
         public bool IsValid()
         {
-            // TODO: Add implementation
+            if (Difficulty == null)
+            {
+                return true;
+            }
+            else if (Difficulty >= (Trail.MinDifficulty ?? 0) && Difficulty <= (Trail.MaxDifficulty ?? Int32.MaxValue))
```

Comment:
The maximum of MaxDifficulty is 1000;  This implementation will fail if Difficulty is greater than 1000 and maxDifficulty is null.  Add a unit test to cover that case, and update your code accordingly.